optimum/exporters/openvino/convert.py

```diff
@@ -0,0 +1,316 @@
+        device (`str`, *optional*, defaults to `cpu`):
+            The device on which the model will be exported. Either `cpu` or `cuda`. Only PyTorch is supported for
+            export on CUDA devices.
+        input_shapes (`Optional[Dict]`, defaults to `None`):
+            If specified, allows to use specific shapes for the example input provided to the exporter.
+
+    Returns:
+        `Tuple[List[str], List[str]]`: A tuple with an ordered list of the model's inputs, and the named inputs from
+        the ONNX configuration.
+    """
+    if not (is_torch_available() or is_tf_available()):
+        raise ImportError(
+            "Cannot convert because neither PyTorch nor TensorFlow are installed. "
+            "Please install torch or tensorflow first."
+        )
+
+    if "diffusers" in str(model.__class__) and not is_diffusers_available():
+        raise ImportError("The pip package `diffusers` is required to export stable diffusion models to ONNX.")
+
+    if is_torch_available() and isinstance(model, nn.Module):
+        return export_pytorch(
+            model,
+            config,
+            opset,
+            output,
+            device=device,
+            input_shapes=input_shapes,
+            model_kwargs=model_kwargs,
+        )
+
+    elif is_tf_available() and issubclass(type(model), TFPreTrainedModel):
+        output.parent.mkdir(parents=True, exist_ok=True)
+        if opset is None:
+            opset = config.DEFAULT_ONNX_OPSET
+        if device == "cuda":
+            raise RuntimeError("`tf2onnx` does not support export on CUDA device.")
+        if input_shapes is not None:
+            logger.info("`input_shapes` argument is not supported by the Tensorflow ONNX export and will be ignored.")
+        return export_tensorflow(model, config, opset, output)
+
+    else:
+        raise RuntimeError(
+            "You either provided a PyTorch model with only TensorFlow installed, or a TensorFlow model with only PyTorch installed."
+        )
+
+
+def export_pytorch_via_onnx(
+    model: Union["PreTrainedModel", "ModelMixin"],
+    config: OnnxConfig,
+    opset: int,
+    output: Path,
+    device: str = "cpu",
+    input_shapes: Optional[Dict] = None,
+    model_kwargs: Optional[Dict[str, Any]] = None,
+):
+    import torch
+
+    output = Path(output)
+    orig_torch_onnx_export = torch.onnx.export
+    torch.onnx.export = functools.partial(orig_torch_onnx_export, do_constant_folding=False)
+    model.config.torchscript = False
+    model.config.return_dict = True
+    onnx_output = output.with_suffix(".onnx")
+    input_names, output_names = export_pytorch_to_onnx(
+        model, config, opset, onnx_output, device, input_shapes, model_kwargs
+    )
+    torch.onnx.export = orig_torch_onnx_export
+    ov_model = convert_model(str(onnx_output))
+    save_model(
+        ov_model,
+        output.parent / OV_XML_FILE_NAME if output.suffix != ".xml" else output,
```

Comment:
should `output_names` be the resulting openvino file (instead of the ONNX one) ? also shouldn't we delete the ONNX model by default ? (adding the possibility to save it if enabled)